build.gradle.kts

```diff
@@ -0,0 +1,348 @@
+tasks.withType<Test> {
+    tasks.getByName("check").dependsOn(this)
+    useJUnitPlatform()
+    testLogging {
+        events("passed", "skipped", "failed")
+    }
+
+    systemProperties(mapOf("org.mongodb.test.uri" to System.getProperty("org.mongodb.test.uri", "")))
+
+    val jdkHome = project.findProperty("jdkHome") as String?
+    jdkHome.let {
+        val javaExecutablesPath = File(jdkHome, "bin/java")
+        if (javaExecutablesPath.exists()) {
+            executable = javaExecutablesPath.absolutePath
+        }
+    }
+
+    addTestListener(object : TestListener {
+        override fun beforeTest(testDescriptor: TestDescriptor?) {}
+        override fun beforeSuite(suite: TestDescriptor?) {}
+        override fun afterTest(testDescriptor: TestDescriptor?, result: TestResult?) {}
+        override fun afterSuite(d: TestDescriptor?, r: TestResult?) {
+            if (d != null && r != null && d.parent == null) {
+                val resultsSummary = """Tests summary:
+                    | ${r.testCount} tests,
+                    | ${r.successfulTestCount} succeeded,
+                    | ${r.failedTestCount} failed,
+                    | ${r.skippedTestCount} skipped""".trimMargin().replace("\n", "")
+
+                val border = "=".repeat(resultsSummary.length)
+                logger.lifecycle("\n$border")
+                logger.lifecycle("Test result: ${r.resultType}")
+                logger.lifecycle(resultsSummary)
+                logger.lifecycle("${border}\n")
+            }
+        }
+    })
+}
+
+/*
+ * Code checking
+ */
+checkstyle {
+    toolVersion = "7.4"
+}
+
+spotbugs {
+    excludeFilter.set(project.file("config/spotbugs/exclude.xml"))
+    showProgress.set(true)
+    setReportLevel("high")
+    setEffort("max")
+}
+
+tasks.withType<com.github.spotbugs.snom.SpotBugsTask> {
+    enabled = baseName.equals("main")
+    reports.maybeCreate("html").isEnabled = !project.hasProperty("xmlReports.enabled")
+    reports.maybeCreate("xml").isEnabled = project.hasProperty("xmlReports.enabled")
+}
+
+// Spotless is used to lint and reformat source files.
+spotless {
+    java {
+        googleJavaFormat()
+        importOrder("java", "io", "org", "org.bson", "com.mongodb", "com.mongodb.spark", "")
+        removeUnusedImports() // removes any unused imports
+        trimTrailingWhitespace()
+        endWithNewline()
+        indentWithSpaces()
+    }
+
+    kotlinGradle {
+        ktlint("0.30.0")
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+
+    format("extraneous") {
+        target("*.xml", "*.yml", "*.md")
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+}
+
+// Auto apply spotless on compile
+tasks.named("compileJava") {
+    dependsOn(":spotlessApply")
+}
+
+/*
+ * ShadowJar
+ */
+tasks.register<ShadowJar>("allJar") {
+    archiveClassifier.set("all")
+    from(mongoDependencies, sourceSets.main.get().output)
```

Comment:
Why do we need to shadow the driver?

Reply:
The Shadow Jar plugin is used to build the uber all jar.